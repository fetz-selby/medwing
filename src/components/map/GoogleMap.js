import React, { Component } from 'react';
import PropTypes from 'prop-types';
import GMap from 'google-map-react';
import {MAP_KEYS, DEFAULT_MAP_CENTER, DEFAULT_MAP_ZOOM} from '../../config';
import Marker from './Marker';

import './google-map.css';

class GoogleMap extends Component {

    loadMarkers = (data) => 
        data.map((location)=>
            <Marker key={location.id} 
                    lat={location.lat} 
                    lng={location.lng} 
                    id={location.id}
                    location={location.address}
                    selected={location.selected} 
                    click={this.onMarkerClickedHandler}/>
        )
    
    onMarkerClickedHandler = (id) =>{
        console.log('id => '+id)
    }

    render() {

        return <div className='map' >
                <GMap 
                    bootstrapURLKeys={{key: MAP_KEYS}}
                    defaultCenter={DEFAULT_MAP_CENTER}
                    defaultZoom={DEFAULT_MAP_ZOOM}
                >

                {this.loadMarkers(this.props.locations)}
                </GMap>
            </div>
    }
}

GoogleMap.propTypes = {
    locations: PropTypes.array.isRequired
}

export default GoogleMap;
