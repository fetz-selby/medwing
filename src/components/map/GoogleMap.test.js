import React from 'react';
import {shallow} from 'enzyme';
import GoogleMap from './';

describe('Map Component', ()=>{
    it('should render maps correctly', ()=>{
        const locations = [{id: 1, lat: '5.5427477', lng: '-0.2565425', address: 'Dansoman, Accra, Ghana', selected: false},
                           {id: 2, lat: '5.7273492', lng: '0.0163231', address: 'Community 25, Tema, Tema, Ghana', selected: false},
                           {id: 3, lat: '5.5367445', lng: '-0.2420576', address: 'Mamprobi, Accra, Ghana', selected: false},
                           {id: 4, lat: '5.5682665', lng: '-0.1717396', address: 'Labone, Accra, Ghana', selected: false}];

        const map = shallow(<GoogleMap locations={locations} keys=''/>);
        expect(map).toMatchSnapshot();
    })
})