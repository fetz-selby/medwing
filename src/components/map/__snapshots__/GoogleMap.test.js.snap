// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map Component should render maps correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GoogleMap
    keys=""
    locations={
      Array [
        Object {
          "address": "Dansoman, Accra, Ghana",
          "id": 1,
          "lat": "5.5427477",
          "lng": "-0.2565425",
          "selected": false,
        },
        Object {
          "address": "Community 25, Tema, Tema, Ghana",
          "id": 2,
          "lat": "5.7273492",
          "lng": "0.0163231",
          "selected": false,
        },
        Object {
          "address": "Mamprobi, Accra, Ghana",
          "id": 3,
          "lat": "5.5367445",
          "lng": "-0.2420576",
          "selected": false,
        },
        Object {
          "address": "Labone, Accra, Ghana",
          "id": 4,
          "lat": "5.5682665",
          "lng": "-0.1717396",
          "selected": false,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <GoogleMap
        bootstrapURLKeys={
          Object {
            "key": "AIzaSyC53deGuD9TB4tJsbB2h_NtOMzUxANOVZs",
          }
        }
        debounced={true}
        defaultCenter={
          Array [
            51.165691,
            10.451526000000058,
          ]
        }
        defaultZoom={5}
        distanceToMouse={[Function]}
        googleMapLoader={[Function]}
        heatmap={Object {}}
        heatmapLibrary={false}
        hoverDistance={30}
        layerTypes={Array []}
        options={[Function]}
        style={
          Object {
            "height": "100%",
            "margin": 0,
            "padding": 0,
            "position": "relative",
            "width": "100%",
          }
        }
        yesIWantToUseGoogleMapApiInternals={false}
      >
        <Marker
          id={1}
          lat="5.5427477"
          lng="-0.2565425"
          location="Dansoman, Accra, Ghana"
          selected={false}
        />
        <Marker
          id={2}
          lat="5.7273492"
          lng="0.0163231"
          location="Community 25, Tema, Tema, Ghana"
          selected={false}
        />
        <Marker
          id={3}
          lat="5.5367445"
          lng="-0.2420576"
          location="Mamprobi, Accra, Ghana"
          selected={false}
        />
        <Marker
          id={4}
          lat="5.5682665"
          lng="-0.1717396"
          location="Labone, Accra, Ghana"
          selected={false}
        />
      </GoogleMap>,
      "className": "map",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bootstrapURLKeys": Object {
          "key": "AIzaSyC53deGuD9TB4tJsbB2h_NtOMzUxANOVZs",
        },
        "children": Array [
          <Marker
            id={1}
            lat="5.5427477"
            lng="-0.2565425"
            location="Dansoman, Accra, Ghana"
            selected={false}
          />,
          <Marker
            id={2}
            lat="5.7273492"
            lng="0.0163231"
            location="Community 25, Tema, Tema, Ghana"
            selected={false}
          />,
          <Marker
            id={3}
            lat="5.5367445"
            lng="-0.2420576"
            location="Mamprobi, Accra, Ghana"
            selected={false}
          />,
          <Marker
            id={4}
            lat="5.5682665"
            lng="-0.1717396"
            location="Labone, Accra, Ghana"
            selected={false}
          />,
        ],
        "debounced": true,
        "defaultCenter": Array [
          51.165691,
          10.451526000000058,
        ],
        "defaultZoom": 5,
        "distanceToMouse": [Function],
        "googleMapLoader": [Function],
        "heatmap": Object {},
        "heatmapLibrary": false,
        "hoverDistance": 30,
        "layerTypes": Array [],
        "options": [Function],
        "style": Object {
          "height": "100%",
          "margin": 0,
          "padding": 0,
          "position": "relative",
          "width": "100%",
        },
        "yesIWantToUseGoogleMapApiInternals": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "click": undefined,
            "id": 1,
            "lat": "5.5427477",
            "lng": "-0.2565425",
            "location": "Dansoman, Accra, Ghana",
            "selected": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "click": undefined,
            "id": 2,
            "lat": "5.7273492",
            "lng": "0.0163231",
            "location": "Community 25, Tema, Tema, Ghana",
            "selected": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "function",
          "props": Object {
            "click": undefined,
            "id": 3,
            "lat": "5.5367445",
            "lng": "-0.2420576",
            "location": "Mamprobi, Accra, Ghana",
            "selected": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "function",
          "props": Object {
            "click": undefined,
            "id": 4,
            "lat": "5.5682665",
            "lng": "-0.1717396",
            "location": "Labone, Accra, Ghana",
            "selected": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <GoogleMap
          bootstrapURLKeys={
            Object {
              "key": "AIzaSyC53deGuD9TB4tJsbB2h_NtOMzUxANOVZs",
            }
          }
          debounced={true}
          defaultCenter={
            Array [
              51.165691,
              10.451526000000058,
            ]
          }
          defaultZoom={5}
          distanceToMouse={[Function]}
          googleMapLoader={[Function]}
          heatmap={Object {}}
          heatmapLibrary={false}
          hoverDistance={30}
          layerTypes={Array []}
          options={[Function]}
          style={
            Object {
              "height": "100%",
              "margin": 0,
              "padding": 0,
              "position": "relative",
              "width": "100%",
            }
          }
          yesIWantToUseGoogleMapApiInternals={false}
        >
          <Marker
            id={1}
            lat="5.5427477"
            lng="-0.2565425"
            location="Dansoman, Accra, Ghana"
            selected={false}
          />
          <Marker
            id={2}
            lat="5.7273492"
            lng="0.0163231"
            location="Community 25, Tema, Tema, Ghana"
            selected={false}
          />
          <Marker
            id={3}
            lat="5.5367445"
            lng="-0.2420576"
            location="Mamprobi, Accra, Ghana"
            selected={false}
          />
          <Marker
            id={4}
            lat="5.5682665"
            lng="-0.1717396"
            location="Labone, Accra, Ghana"
            selected={false}
          />
        </GoogleMap>,
        "className": "map",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bootstrapURLKeys": Object {
            "key": "AIzaSyC53deGuD9TB4tJsbB2h_NtOMzUxANOVZs",
          },
          "children": Array [
            <Marker
              id={1}
              lat="5.5427477"
              lng="-0.2565425"
              location="Dansoman, Accra, Ghana"
              selected={false}
            />,
            <Marker
              id={2}
              lat="5.7273492"
              lng="0.0163231"
              location="Community 25, Tema, Tema, Ghana"
              selected={false}
            />,
            <Marker
              id={3}
              lat="5.5367445"
              lng="-0.2420576"
              location="Mamprobi, Accra, Ghana"
              selected={false}
            />,
            <Marker
              id={4}
              lat="5.5682665"
              lng="-0.1717396"
              location="Labone, Accra, Ghana"
              selected={false}
            />,
          ],
          "debounced": true,
          "defaultCenter": Array [
            51.165691,
            10.451526000000058,
          ],
          "defaultZoom": 5,
          "distanceToMouse": [Function],
          "googleMapLoader": [Function],
          "heatmap": Object {},
          "heatmapLibrary": false,
          "hoverDistance": 30,
          "layerTypes": Array [],
          "options": [Function],
          "style": Object {
            "height": "100%",
            "margin": 0,
            "padding": 0,
            "position": "relative",
            "width": "100%",
          },
          "yesIWantToUseGoogleMapApiInternals": false,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "function",
            "props": Object {
              "click": undefined,
              "id": 1,
              "lat": "5.5427477",
              "lng": "-0.2565425",
              "location": "Dansoman, Accra, Ghana",
              "selected": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "function",
            "props": Object {
              "click": undefined,
              "id": 2,
              "lat": "5.7273492",
              "lng": "0.0163231",
              "location": "Community 25, Tema, Tema, Ghana",
              "selected": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "3",
            "nodeType": "function",
            "props": Object {
              "click": undefined,
              "id": 3,
              "lat": "5.5367445",
              "lng": "-0.2420576",
              "location": "Mamprobi, Accra, Ghana",
              "selected": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "4",
            "nodeType": "function",
            "props": Object {
              "click": undefined,
              "id": 4,
              "lat": "5.5682665",
              "lng": "-0.1717396",
              "location": "Labone, Accra, Ghana",
              "selected": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
